cmd/agent/app/check.go

```diff
@@ -127,19 +129,47 @@ var checkCmd = &cobra.Command{
 			fmt.Println("Multiple check instances found, running each of them")
 		}
 
+		var instancesData []interface{}
+
 		for _, c := range cs {
 			s := runCheck(c, agg)
 
 			// Sleep for a while to allow the aggregator to finish ingesting all the metrics/events/sc
 			time.Sleep(time.Duration(checkDelay) * time.Millisecond)
 
-			printMetrics(agg)
+			if formatJSON {
+				aggregatorData := getMetricsData(agg)
+				var collectorData map[string]interface{}
+
+				collectorJSON, _ := status.GetCheckStatusJSON(c, s)
```

Comment:
This silented error is not dramatic but could lead to the next lines to crash, it may be better to handle them gracefully.